crates/hyperqueue/src/dashboard/ui/screens/home/chart_section.rs

```diff
@@ -0,0 +1,147 @@
+        // Update the number of workers connected
+        let time = get_current_time();
+        let num_workers = overview.worker_overviews.len();
+        if num_workers > self.max_observed_workers as usize {
+            self.max_observed_workers = num_workers as u64;
+        }
+        self.timed_info_vec.push(WorkerStateAtTime {
+            at_time: time,
+            event: TimeEventType::WorkerCountUpdate(num_workers as u32),
+        });
+        // Add more points at the same time to smoothen the transition in plot
+        let interpolation =
+            interpolate_y_axis(self.current_conn_workers as f64, num_workers as f64, 30_f64);
+        for y_point in interpolation {
+            self.timed_info_vec.push(WorkerStateAtTime {
+                at_time: time,
+                event: TimeEventType::WorkerCountUpdate(y_point as u32),
+            })
+        }
+        self.current_conn_workers = num_workers as u64;
+    }
+
+    pub fn draw(&mut self, rect: Rect, frame: &mut DashboardFrame) {
+        let worker_counts: Vec<(f64, f64)> = self
+            .timed_info_vec
+            .iter()
+            .filter_map(|event| {
+                if let TimeEventType::WorkerCountUpdate(count) = event.clone().event {
+                    //fixme: how bad?
+                    Some((event.at_time as f64, count as f64))
+                } else {
+                    None
+                }
+            })
+            .collect();
+
+        let datasets = vec![Dataset::default()
+            .name("workers_connected")
+            .marker(symbols::Marker::Braille)
+            .style(Style::default().fg(Color::White))
+            .data(&worker_counts)];
+
+        let chart = Chart::new(datasets)
+            .block(
+                Block::default()
+                    .title(Span::styled(
+                        "Worker Connection History",
+                        Style::default()
+                            .fg(Color::White)
+                            .add_modifier(Modifier::BOLD),
+                    ))
+                    .borders(Borders::ALL),
+            )
+            .x_axis(
+                Axis::default()
+                    .title("time ->")
+                    .style(Style::default().fg(Color::Gray))
+                    .bounds([self.time_reference as f64, get_current_time() as f64]),
+            )
+            .y_axis(
+                Axis::default()
+                    .style(Style::default().fg(Color::Gray))
+                    .bounds([0.0, self.max_observed_workers as f64])
+                    .labels(vec![
+                        Span::from(0.to_string()),
+                        Span::from((self.max_observed_workers / 2).to_string()),
+                        Span::from((self.max_observed_workers as f64).to_string()),
+                    ]),
+            );
+        frame.render_widget(chart, rect);
+    }
+}
+
+fn get_current_time() -> u128 {
```

Comment:
Let's represent a timestamp in some unified and semantic way. For example `SystemTime` or `Instant`. Not by a "raw" number.

Reply:
what about humantime::Timestamp? can I use it?

Reply:
That's just a wrapper around `SystemTime`, let's just use `SystemTime` directly.